### PR TITLE
signrpc: sign and verify messages with custom key 

### PR DESCRIPTION
**Pull request by guggero**
The existing `SignMessage` and `VerifyMessage` in the main `rpc.proto` file are very restrictive in that they only allow creating signatures over the node's identity private key and that a signature's recovered public key must correspond to a nodes' identity public key known to the network.
The new methods in the `signer.proto` don't have those restrictions and allow signing and verifying over custom wallet keys (limited to the `1017'` special purpose derivation branch).

### CODE REVIEW

File: lnrpc/signrpc/signer_server.go [hunk 1]
```diff
@@ -383,3 +400,75 @@ func (s *Server) ComputeInputScript(ctx context.Context,
...
+	})
+	if err != nil {
+		return nil, fmt.Errorf("can't derive private key: %v", err)
+	}
+
+	// The signature is over the double-sha256 hash of the prefixed message.
+	in.Msg = append(signedMsgPrefix, in.Msg...)
+	digest := chainhash.DoubleHashB(in.Msg)
+
+	// Create the pubkey-recoverable signature using the compressed pubkey.
```
Comment (Roasbeef) [inline lnrpc/signrpc/signer_server.go]: I think for this, we just want a raw regular signature: so no compact encoding, and no `zbase32`, just a raw ECDSA signature.

File: lnrpc/signrpc/signer_server.go [hunk 2]
```diff
@@ -383,3 +400,75 @@ func (s *Server) ComputeInputScript(ctx context.Context,
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode signature: %v", err)
+	}
+
+	// The signature is over the double-sha256 hash of the prefixed message.
+	in.Msg = append(signedMsgPrefix, in.Msg...)
+	digest := chainhash.DoubleHashB(in.Msg)
+
+	// RecoverCompact both recovers the pubkey and validates the signature.
+	pubKey, valid, err := btcec.RecoverCompact(btcec.S256(), sig, digest)
```
Comment (Roasbeef) [inline lnrpc/signrpc/signer_server.go]: Similar message here, at least for the use case we have in mind.

**Comment by guggero** [comment]
I switched to default ECDSA encoding but left the message prefixing in to avoid signing anything sensitive. This means that you can only verify signatures generated with the lnd signer which should be fine for our use case.

I've also given the `VerifyMessage` the `signer/read` permission instead of `signer/generate`.

**Comment by joostjager** [comment]
I think `admin.macaroon` needs to be regenerated? In that case it may be good to add that to the release notes.

**Pull request merged**